drivers/src/bedrock/index.ts

```diff
@@ -349,31 +351,77 @@ export class BedrockDriver extends AbstractDriver<BedrockDriverOptions, BedrockP
         return completion;
     }
 
+    extractRegion(modelString: string, defaultRegion: string): string {
+        // Match region in full ARN pattern
+        const arnMatch = modelString.match(/arn:aws[^:]*:bedrock:([^:]+):/);
+        if (arnMatch) {
+            return arnMatch[1];
+        }
+        
+        // Match common AWS regions directly in string
+        const regionMatch = modelString.match(/(?:us|eu|ap|sa|ca|me|af)[-](east|west|central|south|north|southeast|southwest|northeast|northwest)[-][1-9]/);
+        if (regionMatch) {
+            return regionMatch[0];
+        }
+    
+        return defaultRegion;
+    }
+
+    private async getCanStream(model: string, type: string): Promise<boolean> {
+        let canStream: boolean = false;
+        let error: any = null;
+        const region = this.extractRegion(model, this.options.region);
+        if (type == "foundation-model" || type == "") {
```

Comment:
nit: is it possible to use an `enum` for this? so that we can share the type between caller (canStream) and the implementation of this function

Reply:
Done